### PR TITLE
add process for generating log text analyzers based on collectors present

### PR DESCRIPTION
**Pull request by danbudris**
https://github.com/aws/eks-anywhere/issues/239

## changes
Add system to associate log text analyzers with logs collected from a given namespace.

Move namespace names to `constants` package.

Add initial log text analyzer for `capi-kubeadm-control-plane-system`, which checks for failed API server pods in the logs.

## explanation
Log text analyzers parse the logs collected by the associated Collector for regex matches (or, alternatively, regex group matches -- see https://troubleshoot.sh/docs/analyze/regex/) and produce an outcome based on the result.

We can now add new entires to the `logTextAnalyzersMap` for each namespace; each of the associated analyzers will then be included if the logs for that namespace are being collected. This allows us to compose the log text analyzers based on the collectors that are present; we define the different analyzers that could possibly be present, and then only use the ones that are relevant for the given input cluster. 

This allows us to generate log analyzers dynamically based on the namespaced log collector present; this is useful in cases like `flux-system` or `etcadm-controller-system`, where the logs will not be present if the customer is not using gitops or external etcd. If the customer is using one of these optional features, then we'll be collecting logs for the associated namespace, and will then subsequently add the analyzers for the given logs in the namespace.

## process for adding new text analyzers

to add text analyzers for a given namespace, we'd add an entry to the `logtextAnalyzersMap` for that namespace; the key is the namespace name, the value is a slice of analyzers. Once the entry is in the map, if we collect logs for the namespace, we'll include the analyzer.

to add a new text analyzer to a given namespace, we'd add an additional analyzer to the generator function for that namespace's analyzers -- for example, [add a new analyzer entry to this method for `capi-kubeadm-control-plane-system`](https://github.com/aws/eks-anywhere/pull/334/files#diff-bcb63ea9e7a019b94633aeda38a438f47808688c5e8f5e46775f9958dd6278cdR183).  This allows us to clearly define unique analyzers for each namespace, and control what pod logs in that namespace are analyzed and what the analyzer results return at a granular, per-namespace level.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/334#" title="Author self-approved">danbudris</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [danbudris]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by abhay-krishna** [comment]
@danbudris: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-mocks-presubmits | 96d148578c45b8922b87b08de605461f8980d681 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/334/eks-anywhere-mocks-presubmits/1443579963222528000) | `/test eks-anywhere-mocks-presubmits`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=334). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:danbudris).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by abhay-krishna** [comment]
@danbudris: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-mocks-presubmits | 96d148578c45b8922b87b08de605461f8980d681 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/334/eks-anywhere-mocks-presubmits/1443579963222528000) | `/test eks-anywhere-mocks-presubmits`
eks-anywhere-presubmit | 96d148578c45b8922b87b08de605461f8980d681 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/334/eks-anywhere-presubmit/1443579963251888128) | `/test eks-anywhere-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=334). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:danbudris).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: pkg/diagnostics/analyzers.go
Comment (g-gaston) [inline pkg/diagnostics/analyzers.go:185]: Why not just define this directly in a string literal?

File: pkg/diagnostics/analyzers.go
Comment (g-gaston) [inline pkg/diagnostics/analyzers.go:177]: Seems like all the info coming from this method is static
curious why you decided to have methods instead of "static" vars

**Review by g-gaston — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/334#" title="Author self-approved">danbudris</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/334#pullrequestreview-768375464" title="Approved">g-gaston</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [danbudris,g-gaston]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/diagnostics/analyzers.go
Comment (danbudris) [reply pkg/diagnostics/analyzers.go:185]: do you mean like `fmt.Sprintf("%/manager.log", capiCpManagerPod)`? If so, no real reason, just that it is in fact a path and that seemed sane

if you mean why not just have `log = "capi-kubeadm-control-plane-controller-manager-*/manager.log"`, I wanted to set up a pattern for accessing the logs of containers in pods with multiple containers; in this case, we're not gathering the logs for the rbacproxy container so it does not really matter, but if we wanted to add it it the pod name is already seperated

File: pkg/diagnostics/analyzers.go
Comment (danbudris) [reply pkg/diagnostics/analyzers.go:177]: my thought was it would be easier to add logic to the generation of the text analyzers in the future if necessary if they were struct methods

**Review by g-gaston — COMMENTED** [review]
/lgtm

**Pull request merged**
